README.md

```diff
@@ -17,173 +20,137 @@ Your test directory should look like this:
-| `bearer`     | Not applicable for `bearer` authorization                                  |
+The generated test uses the `seekret` fixture, which returns an instance of the `seekret.apitest.Context` class. The
+test context allows separating test stages with the `stage` context manager, and sending requests to the target server.
+The context uses the information from the [run profile](#run-profile) in order to determine the target server, and the
+required authorization data.
 
-### Example: Login with credentials and get bearer token
+You can choose the user for a request with by specifying the `user` parameter in the `request()` call to another user
+from the [run profile](#run-profile).
 
-This example describes the authorization stage of a login requests that generates a bearer token.
+### Setup and teardown
 
-```yaml
-variables:
-  seekret:
-    v1:
-      target_server: http://example.com
-      users:
-        user:
-          auth:
-            type: custom-request
-            data:
-              auth_stage_id: login
-              username: Seekret
-              password: isAwesome!
-
-stages:
-  - id: login
-    name: Login
-    request:
-      method: POST
-      url: /login
-      json:
-        username: '{seekret.v1.users.auth.data.username}'
-        password: '{seekret.v1.users.auth.data.password}'
-    response:
-      status_code:
-        - 201
-      save:
-        $ext:
-          function: seekret.apitest:save_authorization
-          extra_kwargs:
-            json: data.token
-            type: bearer
-```
\ No newline at end of file
+You can add setup and teardown logic to tests
+using [pytest fixtures](https://docs.pytest.org/en/stable/fixture.html#fixture). The `seekret` fixture is available for
+use in function-scoped fixtures for declaring setup stages and logging.
+
+Consider the following example of setup and teardown logic:
+
+```python
+import pytest
+import seekret.apitest
+
+
+@pytest.fixture
+def channel(seekret: seekret.apitest.Context):
+    # Setup stage: happens before the test.
+    with seekret.stage(method='POST', path='/api/channels') as request:
+        response = request(json={
+            'name': 'fuGfYxcrgCSsVBMLlceEpLrGPT'
+        })
+        assert response.status_code == 201
+
+        channel_id = response.search('json.data.channel.id')
+
+    # Yielding runs the actual test body.
+    # Important! Yield outside of the stage context. Otherwise the test will start in the context of the setup stage.
+    yield channel_id
+
+    # Teardown stage: happens after the test.
+    with seekret.stage(method='DELETE', path='/api/channels/{channel_id}'):
+        response = request(path_params={
+            'channel_id': channel_id
+        })
+        assert response.status_code == 200
+
+
+def test_post_messages_delete_messages(seekret: seekret.apitest.Context, channel):
+    # The 'channel' parameter has the value yielded from the 'channel' fixture.
```

Comment:
expand the example here, do they use `channel` or `channel_id`

Reply:
Fixed